mPower2/mPower2/WithdrawalViewController.swift

```diff
@@ -86,55 +86,69 @@ class WithdrawalViewController: UIViewController {
     
     @objc
     func withdrawalTapped() {
-        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate,
-        let consentStatuses = appDelegate.userSessionInfo?.consentStatuses as? [ String : SBBConsentStatus ]
+        guard let withdrawalSurvey : SBBSurveyReference = SBABridgeConfiguration.shared.allSurveys().first(where: { (ref) -> Bool in
+                return ref.identifier == .withdrawal
+            })
             else {
                 return
         }
         
         // Hide the back button so we can be sure to still be around when the response alert pops up
         self.backButton.isHidden = true
         
-        // Withdraw them from all subpopulations whose current consent they have signed.
-        let withdrawGroup = DispatchGroup()
-        var errorWithdrawing: Bool = false
-        for (subpopulationGuid, consentStatus) in consentStatuses {
-            if consentStatus.signedMostRecentConsentValue && !errorWithdrawing {
-                withdrawGroup.enter()
-                BridgeSDK.consentManager.withdrawConsent(forSubpopulation: subpopulationGuid, withReason: "chose to withdraw from app via profile tab") { (response, error) in
+        // Present the withdrawal survey to get their reasons, and withdraw (or not) based on their completion (or not) of the survey
+        let vc = RSDTaskViewController(taskInfo: withdrawalSurvey)
+        vc.delegate = self
+        self.present(vc, animated: true)
+    }
+    
+    // MARK: RSDTaskViewControllerDelegate
+    func taskController(_ taskController: RSDTaskController, didFinishWith reason: RSDTaskFinishReason, error: Error?) {
+        guard let appDelegate = UIApplication.shared.delegate as? AppDelegate else { return }
+        let reasonsAnswer = taskController.taskResult.findAnswerResult(with: "withdrawalReason")
+        (taskController as! UIViewController).dismiss(animated: true) {
+            if reason == .completed {
```

Comment:
Nit: use a guard and early exit rather than nested curly brackets.

```
guard reason == .completed else { return }
```

Reply:
whoops, forgot the else clause